test/khash_test.erl

```diff
@@ -0,0 +1,443 @@
+-define(NUM_CYCLES, 1000000).
+-define(NUM_KVS, 5000).
+
+load_test_() ->
+    {
+        "Loaded khash",
+        ?_assertEqual({module, khash}, code:load_file(khash))
+    }.
+
+basic_test_() ->
+    {
+        "khash basic operations",
+        {setup,
+            local,
+            fun() -> khash:new() end,
+            fun({ok, _}) -> ok end,
+            fun({ok, C}) ->
+                [
+                    {
+                        "Lookup missing is ok",
+                        ?_assertEqual(not_found, khash:lookup(C, <<"foo">>))
+                    },
+                    {
+                        "Get missing is ok",
+                        ?_assertEqual(undefined, khash:get(C, <<"foo">>))
+                    },
+                    {
+                        "Del missing is ok",
+                        ?_assertEqual(not_found, khash:del(C, <<"foo">>))
+                    },
+                    {
+                        "Stored a key",
+                        ?_assertEqual(ok, khash:put(C, <<"foo">>, bar))
+                    },
+                    {
+                        "Lookuped a key",
+                        ?_assertEqual({value, bar}, khash:lookup(C, <<"foo">>))
+                    },
+                    {
+                        "Retrieved a key",
+                        ?_assertEqual(bar, khash:get(C, <<"foo">>))
+                    },
+                    {
+                        "Stored a key",
+                        ?_assertEqual(ok, khash:put(C, <<"bar">>, foo))
+                    },
+                    {
+                        "Correct size for hash",
+                        ?_assertEqual(2, khash:size(C))
+                    },
+                    {
+                        "Deleted a key",
+                        ?_assertEqual(ok, khash:del(C, <<"foo">>))
+                    },
+                    {
+                        "Correct size after delete",
+                        ?_assertEqual(1, khash:size(C))
+                    },
+                    {
+                        "Cleared the hash",
+                        ?_assertEqual(ok, khash:clear(C))
+                    },
+                    {
+                        "Correct size after clear",
+                        ?_assertEqual(0, khash:size(C))
+                    }
+                ]
+            end
+        }
+    }.
+
+randomized_test_() ->
+    {
+        "khash randomized test",
+        {setup,
+            local,
+            fun() ->
+                random:seed(erlang:now()),
+                Dict = dict:new(),
+                {ok, KHash} = khash:new(),
+                Actions = [
+                    {0.1, fun run_clear/1},
+                    {1.0, fun run_get2/1},
+                    {1.0, fun run_get3/1},
+                    {1.0, fun run_put/1},
+                    {1.0, fun run_del/1},
+                    {0.5, fun run_size/1},
+                    % {0.3, fun run_keys/1},
```

Comment:
I scanned the lib's source and it seems that `khash:keys/1` is planned, but not implemented feature. My guess is that this is sort of a placeholder/reminder.